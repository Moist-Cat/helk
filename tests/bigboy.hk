type Aleph {
    x = 0;
    y = "set";

    constant() => 1000;

    do(x) => self.x;
};

type AlephNull inherits Aleph {
    z = 10;

    do(x) => self.x + self.z + self.constant() + x;
};

function gcd(a, b) => while (a) {
    let m = a % print(b), b = a, a = m in a;
};

function f(a, b) => while (a) {
    let a = a - 1 in print(a);
};

function fib(n) =>
    if (min(max(n - 2, 0), 1)) {
        fib(n - 1) + fib(n - 2);
    }
    else {
        1;
    };

prints("Conflagration of all things");
let d = 10;
let e = 10;
let aleph = new AlephNull(10, "dummy", 10);
print(aleph.do(5));
while (d) {
    let d = d - 1;
    let e = e - 10;
    if (min(max(d - 5, 0), 1)) {
        prints("Scoping");
        let a = 10 in while(0) {
            let a = 5 in print(a);
            print(a);
        };
        prints("Fib:");
        print(fib(d));
        prints("Aleph:");
        print(aleph.do(e));
    }
    else {
        print(1 + 1 + 1);
        prints("GCD:");
        print(gcd(e, d));
    };
};
print(e);
