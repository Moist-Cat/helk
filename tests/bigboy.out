Conflagration of all things
1025.000000
Scoping
5.000000
10.000000
Fib:
34.000000
Aleph:
1020.000000
Scoping
5.000000
10.000000
Fib:
21.000000
Aleph:
1010.000000
Scoping
5.000000
10.000000
Fib:
13.000000
Aleph:
1000.000000
Scoping
5.000000
10.000000
Fib:
8.000000
Aleph:
990.000000
3.000000
GCD:
5.000000
-40.000000
0.000000
3.000000
GCD:
4.000000
-50.000000
-2.000000
-2.000000
0.000000
3.000000
GCD:
3.000000
-60.000000
0.000000
3.000000
GCD:
2.000000
-70.000000
0.000000
3.000000
GCD:
1.000000
-80.000000
0.000000
3.000000
GCD:
0.000000
-90.000000
0.000000
3.000000
GCD:
-1.000000
-100.000000
0.000000
-90.000000
